source/common/upstream/cluster_manager_impl.cc

```diff
@@ -831,6 +831,15 @@ ClusterManagerImpl::loadCluster(const envoy::config::cluster::v3::Cluster& clust
     }
   } else if (cluster_reference.info()->lbType() == LoadBalancerType::ClusterProvided) {
     cluster_entry_it->second->thread_aware_lb_ = std::move(new_cluster_pair.second);
+  } else if (cluster_reference.info()->lbType() == LoadBalancerType::LoadBalancingPolicyConfig) {
+    const auto& policy = cluster_reference.info()->loadBalancingPolicy();
+    TypedLoadBalancerFactory* typed_lb_factory =
+        Registry::FactoryRegistry<TypedLoadBalancerFactory>::getFactory(policy.name());
+    RELEASE_ASSERT(typed_lb_factory != nullptr, "should have factory for selected LB policy");
```

Comment:
use `Config::Utility::getAndCheckFactoryByName` to reject config correctly instead of RELEASE_ASSERT.

Reply:
I think `typed_lb_factory` being `nullptr` is actually an impossible condition here. We don't need to validate that the factory exists because the `ClusterInfoImpl` constructor should have already done that - it throws if it doesn't find any factory matching a LB policy specified in `load_balancing_policies`.